utils/randomize.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+# use a POSIX shell shuf alternative if none on the system
+command -v shuf > /dev/null 2> /dev/null || shuf() {
+    awk '
+        BEGIN {
+            srand();
+            OFMT = "%.17f"
+        }
+        {
+            print rand(), $0
+        }
+    ' "$@" | 
+    sort -k1,1n | 
+    cut -d ' ' -f2- ;}
+
+# first argument is the filename
+filename="$1"
+[ ! -f "$filename" ] && printf "%s\n" "You must provide a text file as argument." >&2 && exit 1
+
+words=""
+# read file line by line
+while IFS= read -r line; do
+    # split the line into words and set them as positional parameters
+    # shellcheck disable=SC2086
+    set -- $line
+    for val; do
+        # clean up each word: remove leading and trailing punctuation and quotes
+        val=$(printf "%s" "$val" | sed -e 's/^[({'"'"'"]*//' -e 's/[.,!?;:)]}'"'"'"]*$//')
```

Comment:
I think this would do the same thing as this cryptic sed command and better, but I will be honest, I haven't tried the punctuation cleaning feature with eider commands.
```sh
# delete leading and trailing punctuation and whitespace from each line
sed -e 's/^[[:punct:][:space:]]*//' -e 's/[[:punct:][:space:]]*$//'
```